### PR TITLE
[NFC] Refactor EZPreferencesWindowController into Swift

### PR DESCRIPTION
**Pull request by Kyle-Ye**
A base PR to migrate Preferences System into Swift.

Also [MASPreferences](https://github.com/shpakovski/MASPreferences) is a very old framework written by ObjectiveC and it has been archived by the owner on Mar 6, 2023.

Can we consider migrating into other modern Swifty framework? eg. [Settings](https://github.com/sindresorhus/Settings)

If you agree, I'd like to split the migration into 4 steps. And everyone can do one if they had time.

1. Migrate EZPreferencesWindowController from ObjectiveC into Swift
2. Migrate other MASPreferencesViewController into Swift (`Settings` does not support ObjectiveC)
3. Migrate from  MASPreferences -> Settings (MASPreferencesViewController -> SettingsPane)
> I'll also make the Preference UI part as a standalone SwiftPM to reduce the xcodeproj dependency.
4. (Optional) Write some PreferencesViewController using SwiftUI instead of AppKit in the future
> `Settings` supports AppKit, SwiftUI and mix them together.

image

### CODE REVIEW

**Comment by Kyle-Ye** [comment]
干掉 MASPreferences 后就不用 Easydict 自己维护一套 DarkManager 了，可以直接走 Color Asset 就行。
目前 MASPreferences 下使用这套看起来会有 bug 😂

**Comment by tisfeng** [comment]
可以的，之前我想重构设置页时也看过 [Settings](https://github.com/sindresorhus/Settings) ，只是担心一下改动太大，不好维护，现在有你来负责 Swift 改造自然是好的 😃

用 SwiftUI 也是可以的，只是重构尽量和之前保持一致，一步步来，保持平稳过渡。

前期我可能没办法给到帮助，Swift 部份代码需要你来维护，辛苦了。

File: Easydict.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -2365,7 +2363,7 @@
 				ALWAYS_EMBED_SWIFT_STANDARD_LIBRARIES = YES;
 				BUNDLE_LOADER = "$(TEST_HOST)";
 				CLANG_CXX_LANGUAGE_STANDARD = "gnu++20";
-				CODE_SIGN_IDENTITY = "$CODE_SIGN_IDENTITY";
+				CODE_SIGN_IDENTITY = $CODE_SIGN_IDENTITY;
```
Comment (tisfeng) [inline Easydict.xcodeproj/project.pbxproj:2366]: 你好，请问这里是什么改动？

**Review by tisfeng — CHANGES_REQUESTED** [review]

File: Easydict.xcodeproj/project.pbxproj (hunk 1 above)
Comment (Kyle-Ye) [reply Easydict.xcodeproj/project.pbxproj:2366]: 之前 PR 的 CODE_SIGN_IDENTITY = xx 是手写修改的，习惯性加上 "" 防止空格转义，这里在 Xcode 15 下打开触发了该文件的Xcode内部编辑（添加删除了部分文件）后自动被 format 去掉了。

就这样改下吧，不然后续其他人每次增删文件后都会有这个 diff

**Comment by Kyle-Ye** [comment]
> 用 SwiftUI 也是可以的，只是重构尽量和之前保持一致，一步步来，保持平稳过渡。

Sure. Let's do it step by step. 🚀

**Review by tisfeng — APPROVED** [review]

**Pull request merged**